var/spack/repos/builtin/packages/ceres-solver/package.py

```diff
@@ -25,12 +26,25 @@ class CeresSolver(CMakePackage):
     variant("shared", default=True, description="Build shared libraries")
     variant("examples", default=False, description="Build examples")
 
+    depends_on("cmake@2.8.0:", type="build", when="@1.12.0:1.14.0")
+    depends_on("cmake@3.5:", type="build", when="@2.0.0")
+    depends_on("cmake@3.16:3.27", type="build", when="@2.2.0")
     depends_on("eigen@3:")
+    depends_on("eigen@3.3:", when="@2.0.0:")
     depends_on("lapack")
-    depends_on("glog")
+    depends_on("glog@0.3.5:")
 
     def cmake_args(self):
-        args = ["-DCXSPARSE=OFF", "-DEIGENSPARSE=ON", "-DLAPACK=ON", "-DSCHUR_SPECIALIZATIONS=OFF"]
+        args = []
+        if self.spec.satisfies("@:2.0.0"):
```

Comment:
Curious. 

Why are the rest of these options no longer relevant starting with v2.2.0 (aside from perhaps `CXSPARSE`) or are you relying on defaults?

Are the docs outdated at http://ceres-solver.org/installation.html#options-controlling-ceres-configuration?

Reply:
CXSPARSE is no longer present in the cmake config, everything else defaults to ON. I don't understand why SCHUR_SPECIALIZATIONS was being disabled, but in the interest of not changing the behavior of 2.0.0 and below I decided to preserve build behavior for the older versions. What prompted the changes originally was a failure to build 2.2.0 at least +suitesparse (which comes highly recommended, enough that I was tempted to make the variant default=True).